.ci/set_dependency_version

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [ -z "${MAIN_REPO_DIR:-}" ]; then
+  export MAIN_REPO_DIR="$(readlink -f "$(dirname "${0}")/..")"
+fi
+
+echo "repo-dir: ${MAIN_REPO_DIR}"
+
+if [ "${DEPENDENCY_NAME}" != "github.com/gardener/machine-controller-manager" ]; then
+  echo "error: do not know how to upgrade ${DEPENDENCY_NAME}"
+  exit 1
+fi
+
+if [ -z "${DEPENDENCY_VERSION:-}" ]; then
+  echo "env var DEPENDENCY_VERSION must be set"
+  exit 1
+fi
+
+# for now, we only know how to upgrade github.com/gardener/machine-controller-manager, as checked above
+
+MCM_FILEPATH="${MAIN_REPO_DIR}/MCM_VERSION"
+
+if [ ! -f "${MCM_FILEPATH}" ]; then
+  echo "error: no such file: {MCM_FILEPATH}"
+  exit 1
+fi
+
+echo -n "${DEPENDENCY_VERSION}" > "${MCM_FILEPATH}"
+echo "set dependency-version of ${DEPENDENCY_NAME} to ${DEPENDENCY_VERSION}"
```

Comment:
Script which will be called by the upgrade_pr job in the pipeline. It checks whether the upgraded component is the mcm component and writes the new version in a file. You can extend the script with your own logic or call other scripts. The complete file system diff at the end of the script will be committed.
https://gardener.github.io/cc-utils/traits/update_component_deps.html#component-upgrade-contract

Reply:
Thank you so much for this PR @jkrayl Loved this! 

I would also want the automated PR to include the vendored files from that new version of MCM. I have extended the bash script as you suggested to the achieve the same. Could you review it once and if you are convinced that the new set of commands would work then please add the suggested commit to this PR?

```suggestion
echo "set dependency-version of ${DEPENDENCY_NAME} to ${DEPENDENCY_VERSION}"

cd ${MAIN_REPO_DIR}
old_version=$(cat go.mod | grep github.com/gardener/machine-controller-manager | xargs)
new_version="github.com/gardener/machine-controller-manager ${DEPENDENCY_VERSION}"
sed -i -- 's#'"${old_version}"'#'"${new_version}"'#g' go.mod

make revendor
```

Reply:
@jkrayl Any view or comments on this?

Reply:
lgtm :) 
I haven't tested it but the changed files will be committed at the end of the job so this will work

Reply:
Thank you, Johannes for the response. I will test it out with the idea you suggested.